browser/src/control/Parts.js

```diff
@@ -487,9 +487,26 @@ L.Map.include({
 				}
 			};
 
-			this.uiManager.showInfoModal('show-sheets-modal', '', ' ', ' ', _('Close'), callback, true, 'show-sheets-modal-response');
+			this.uiManager.showInfoModal('show-sheets-modal', _('Show sheets'), ' ', ' ', _('OK'), callback, true, 'show-sheets-modal-response');
 			const modal = document.getElementById('show-sheets-modal');
 			modal.insertBefore(container, modal.children[0]);
+
+			var okButton = document.getElementById('show-sheets-modal-response');
+			var checkboxes = document.querySelectorAll('#show-sheets-modal input[type="checkbox"]');
```

Comment:
Above you have "callback" which is reponsible for handling user interactions with JSDialog widgets. It can have few parameters which points which widget was modified (see https://github.com/CollaboraOnline/online/blob/b5b70c09046ff2b0d4fcd65c202a3f22bf2b7940/browser/src/control/jsdialog/Definitions.Types.ts#L63).
Code should be moved there.

(Best would be to use JSDialog events to not break the states synchronization, but at least please put the code there)

Reply:
Thanks @eszkadev for the feedback! I understand the suggestion to move the code to align with the callback handling and use JSDialog events for better state synchronization. However, I'm not entirely clear on how to implement this without breaking state synchronization.

Could you please clarify:
1. How the JSDialog events should be used to avoid breaking state synchronization?
2. How the changes should align with the provided Definitions.Types.ts reference

I want to make sure the update is done correctly, Thanks for your help!

Reply:
ad2. Definitions.Types.ts provides you callback type - so you knwo what types of params are there.
ad1. let's ignore JSDialog thing for now, just please put code into callback: example - https://github.com/CollaboraOnline/online/blob/master/browser/src/control/Control.ServerAuditDialog.ts#L270

Reply:
to disable enable button you can use helper here: https://github.com/CollaboraOnline/online/blob/master/browser/src/control/jsdialog/Util.ModalHelper.js#L113

Reply:
```
const modalId = JSDialog.generateModalId('show-sheets-modal');
JSDialog.enableButtonInModal(modalId, modalId + '-response', false);
```